### PR TITLE
detect ssh public key errors and give some debugging information

### PR DESCRIPTION
**Pull request by ricardochimal**
Detect ssh public key errors so that we can give some reasonable debugging information.

### CODE REVIEW

**Comment by ricardochimal** [comment]
I'm not entirely sure how to run the tests locally so any help is appreciated.  `npm test` failed on the master branch on my machine.

File: lib/dyno.js [hunk 1]
```diff
@@ -179,27 +179,52 @@ class Dyno extends Duplex {
...
-        stdio: 'inherit'
+      // only intercept the stderr output
+      let sshProc = spawn('ssh', params, {
+        stdio: [process.stdin, process.stdout, 'pipe']
+      })
+      sshProc.stderr.on('data', (data) => {
+        lastErr = data
+
+        // supress host key and permission denied messages
+        if (this._isDebug() || (data.indexOf("Warning: Permanently added '[127.0.0.1]") === -1 && data.indexOf("Permission denied (publickey).") === -1)) {
```
Comment (jdxcode) [inline lib/dyno.js]: you can use `data.includes()` to make these a bit more concise

File: lib/dyno.js [hunk 2]
```diff
@@ -179,27 +179,52 @@ class Dyno extends Duplex {
...
+           process.stderr.write(data)
+        }
+      })
+      sshProc.on('close', (code, signal) => {
+        // there was a problem connecting with the ssh key
+        if (lastErr.length > 0 && lastErr.indexOf('Permission denied') !== -1) {
+          cli.error("There was a problem connecting to the dyno.")
+          if (process.env.SSH_AUTH_SOCK && process.env.SSH_AUTH_SOCK.length > 0) {
```
Comment (jdxcode) [inline lib/dyno.js]: in javascript you can just do `if (process.env.SSH_AUTH_SOCK)` (`'' === true` in js)

**Comment by jdxcode** [comment]
@ricardochimal you need to use yarn so it respects the lockfile

**Pull request merged**